apps/phone/src/apps/twitter/components/tweet/TweetList.tsx

```diff
@@ -1,21 +1,78 @@
-import { memo, useState } from 'react';
-
-import { List } from '@ui/components/List';
+import { memo, useCallback, useState } from 'react';
 import Tweet from './Tweet';
-import { FormattedTweet } from '@typings/twitter';
+import { Virtuoso } from 'react-virtuoso';
+import { FormattedTweet, GetTweet, TwitterEvents } from '@typings/twitter';
 import Backdrop from '@ui/components/Backdrop';
+import { ServerPromiseResp } from '@typings/common';
+import fetchNui from '@utils/fetchNui';
+import { processTweet } from '@apps/twitter/utils/tweets';
+import { usePhone } from '@os/phone/hooks';
 
 export function TweetList({ tweets }: { tweets: FormattedTweet[] }) {
+  const { ResourceConfig } = usePhone();
+
+  const TWEET_LIMIT_PER_PAGE  = ResourceConfig?.twitter?.resultsLimit || 25;
   const [imageOpen, setImageOpen] = useState<string | null>(null);
+  const [tweetsData, setTweetsData] = useState(tweets);
+  const [hasNextPage, setHasNextPage] = useState(tweetsData.length == TWEET_LIMIT_PER_PAGE ? true : false);
+  const [page, setPage] = useState(0);
+  const [isNextPageLoading, setIsNextPageLoading] = useState(false);
+
+  const handleNewPageLoad = useCallback(() => {
+    if (isNextPageLoading || !hasNextPage) return;
+    setIsNextPageLoading(true);
+
+    const pageId = page + 1
+    fetchNui<ServerPromiseResp<GetTweet[]>>(TwitterEvents.FETCH_TWEETS, {pageId}).then((resp) => {
+      if (resp.status !== 'ok') {
+        console.log(resp.errorMsg || '')
```

Comment:
We should display an error snackbar here, instead of just logging it.

Reply:
done 👍